### ISSUE TITLE
Highlight tiles DK1 style with 'Right mouse button'.

### ISSUE THREAD

**Issue by Loobinex** [OpenDungeons/OpenDungeons#1132]
The games allows you to highlight tiles for digging by dragging a bounding box over the dirt tiles. This is very convenient when making rectangular rooms, but very inconvenient when you want to make corridors or L shaped rooms.

I would like to be able to highlight (and remove highlights from) dirt tiles 'dungeon keeper 1' style by using the right mouse button. So:
- Press right mouse button on a dirt tile to highlight it, and drag your mouse over other tiles to highlight them all.
- Press right mouse button on a highlighted dirt tile to remove the highlight, and drag to remove highlights from others.

**Comment by Bertram25** [comment]
Hi again @Loobinex 

Btw, do you have a video of this DK1 feature somewhere?

As for corridors, it is already feasible by dragging from one tile to another in a row or columns, is there another need in that area?
I don't understand also the need for L shaped rooms, I must say as I never used that in the many games we had.

> ```
> Press right mouse button on a dirt tile to highlight it, and drag your mouse over other tiles to highlight them all.
> Press right mouse button on a highlighted dirt tile to remove the highlight, and drag to remove highlights from others.
> ```

Sorry again, XD, what you describe is the exact current game behaviour using left mouse button.
Could you precise exact use cases?

(Maybe it's time I play DK1 again XD)

**Comment by Bertram25** [comment]
Arf, I think there is a good example here at 0:40: https://www.youtube.com/watch?v=wLzBb2BToZI

I think this may be easily feasible, but I'd rather use some key like Shift or something equilvalent.
@OpenDungeons/developers WDYT all?

**Comment by Loobinex** [comment]
Current implementation is 'selecting' an area to highlight. DK1 functionality is 'painting' an area to highlight. Both have advantages and disadvantages, and having both options provides the best of both world. Highlighting is very easy for making rooms, but frustrating when you want custom shapes and when perspective is off. Paining is slow for large rooms but satisfying to do and easy for custom shapes.

My guess is that both would be used a lot, and personally I would not put it under shift/ctrl command.

**Comment by hwoarangmy** [comment]
I wouldn't do that for building rooms (even if it can be handy especially for treasure room) because when a room is selected, right clic is used to cancel selection.
However, we can use that to select tiles to dig. It wouldn't go against our current workflow. Moreover, our architecture is already more or less prepared for that. In any case, I am planning on improving it so that this case of tile selection becomes very straight forward.

But we are near a release and it won't happen for next release ;-)

@Loobinex Which version are you playing ? What did you think of the game ?

**Comment by Loobinex** [comment]
@hwoarangmy I played the latest version available on opendungeons.github.io, 0.6.0. I haven't prepared anything to build the game myself and have not found any pre-build incremental patches to nightly builds to look at.

I'm still adjusting to the game, just touched it for a few moments. During the daytime I couldn't really play as even with my monitor brightness maxed out it was too dark to see what was going on. And as there's very little player feedback implemented in this build it will take some time to learn what's going on when playing. There's also some usability issues that I first have to learn to cope with, but I'm not yet up to speed on what's already fixed in more recent builds and what's already on the roadmap.

I do certainly appreciate what has been accomplished so far. I'll probably contribute some more after I get up to speed with the project, as I'm a fan of Dungeon Keeper, have some relevant experience, time available, and KeeperFX is on a little hiatus.

**Comment by Danimal696** [comment]
so its a choosing of rigth or left key? i dont think that justifies a change; also DK digging system was pretty bad, you had to mark tiles one by one, that was corrected in DK2 which system we (and many ones) copied. Rigth now you can also select tile by tile or mark rectangles , Whats the problem?

**Comment by akien-mga** [comment]
You can grab newer builds for Windows and Linux on http://download.tuxfamily.org/opendungeons/WIP/

**Comment by Loobinex** [comment]
In DK1 you did not have to select tiles one by one. You could keep your mouse button pressed to draw selection. For small and medium sized rooms it went about as fast as in DK2. 

Room tiles in DK1 did have to be placed one by one though, which even though is too archaic to be re- implemented in a new game so I think you should keep this part as it is. However it did also have a specific advantage over the DK2 implementation and the effect on the player must be understood:
In DK2 and in OpenDungeons the player indicates where the room has to go and the room magically appears. The sounds and animations of DK2 enforce this feeling.
In DK1 the player has the feeling of building the room himself. Each tile is plunked down with force by the player. As a result building rooms in DK1 is more engaging, which is not unimportant in a dungeon building sim.

**Comment by hwoarangmy** [comment]
@Loobinex As I said, we are near a new release and pretty much things have been added since 0.6.0 is out. If you like the game, I strongly recommand to download one of the latest snapshots as many game content as been added. However, the game brightness will probably not be much better as IIRC, nothing has been changed about that. For my part, I don't have this issue so it is not that easy to fix it. I think there is an issue to change the overall lightning to allow to change brightness.

If you want to contribute, you are welcome. You can find some information about how to compile in the wiki (assuming you want to contribute on code):
https://github.com/OpenDungeons/OpenDungeons/wiki/Compile

And if you have problems/questions, do not hesitate to drop on the forum or to open an issue here.

When you will have played a bit, please feel free to give us some feedback as it is nice to know how the game feels for new players. Most of us have been playing it for a while and we sometimes miss some stuff we got used to. But as a former DK player, you should not be lost for too long.

**Comment by Loobinex** [comment]
I ran into another use case for this request.

Play the map Stonekeep Siege and you're surrounded by conveniently placed gold that you'll receive as you dig out rooms. However, proceed after that and you'll need to mine out irregular shaped gold, which is annoying to highlight with the current method and I find myself just selecting square patches and have my workers dig out the surrounding soil.

@hwoarangmy A bit off topic, I've played a bit and I can't tell your direction yet. Do you have some high level design specs, I can't find them on the wiki or the website.

**Comment by hwoarangmy** [comment]
> Do you have some high level design specs

Do you mean about the game play overall ? Or code architecture ?

**Comment by Loobinex** [comment]
Purely game play. And I'm not talking about the specifics like which 
rooms/creatures are to be implemented or what the setting is to be(evil 
creatures/demons), but the kind of stuff you'd normally test in your 
first vertical slice. What is going to engage the player, why is he 
going to play the game, and what is going to provide the fun?

**Comment by Danimal696** [comment]
those are pretty mudddy waters what you are asking for, i can see different answers, but firstly have into account this game is a work in progress. What to expect would be basically the same as DK:
- Dungeon building
- RTS and creature simulator
- Good humour (not yet in)
- Some fun campaign

**Comment by Loobinex** [comment]
A work in progress sure, but I'm understanding you're not working towards a design? Did you receive any input from some experienced (game) designers?

**Comment by Danimal696** [comment]
No, we dont have that kind of budget, but we do have an overall design, played&tested mechanics and most importly, the will to finish this as best as some hobbist ragtag team can. Thats the reason we love getting feedback, it help us improve; but we wont deviate much from our main roadmap to cater to everyone tastes. Thats is a certified project killer.

**Comment by akien-mga** [comment]
There are are no strong design plans at the moment no, I don't know if there were when the original developers started OpenDungeons, or if they just wanted to implement a DK-like engine. When we took over the development, we went from an alpha graphics engine to something that actually looks like a game - @Danimal696 was the guy with the clearest view of the design we should aim for I think, but it mostly evolved little by little as we added what we wanted to add and discussed how it should impact the gameplay.

So of course, it's pretty clear when playing the game that it was development to match a well structured design; we mostly build it so that we can have some fun playing it, and hope to get player feedback to see what direction to go on. So far we plan milestones which are mostly based on necessary features (rooms, creature behaviours, spells, AI, networking, scripting, campaigns, etc.).

Now, I also think that we need a clearer view of what we want the "final" game to be like, and to have answers to the basic questions you ask such as "what is going to engage the player" and "where does the fun come from". If you want to help with that, it would be very welcome :)
